src/initialize_dae.jl

```diff
@@ -36,36 +40,52 @@ end
 
 function _initialize_dae!(integrator, prob::ODEProblem,
                           alg::DefaultInit, x::Val{true})
+    isad = SciMLBase.forwarddiffs_model(integrator.alg)
+    nlsolve = NewtonRaphson(autodiff=isad)
     _initialize_dae!(integrator, prob,
-                     BrownFullBasicInit(integrator.opts.abstol), x)
+                     BrownFullBasicInit(integrator.opts.abstol,nlsolve), x)
 end
 
 function _initialize_dae!(integrator, prob::ODEProblem,
                           alg::DefaultInit, x::Val{false})
+    isad = SciMLBase.forwarddiffs_model(integrator.alg)
+    if integrator.u isa Number || integrator.u isa StaticArray
+        nlsolve = SimpleNewtonRaphson(autodiff=isad)
+    else
+        nlsolve = NewtonRaphson(autodiff=isad)
+    end
+
     _initialize_dae!(integrator, prob,
-                     BrownFullBasicInit(integrator.opts.abstol), x)
+                     BrownFullBasicInit(integrator.opts.abstol, nlsolve), x)
 end
 
 function _initialize_dae!(integrator, prob::DAEProblem,
                           alg::DefaultInit, x::Val{false})
+    isad = SciMLBase.forwarddiffs_model(integrator.alg)
+    if (integrator.u isa Number && integrator.du isa Number) ||
+       (integrator.u isa StaticArray && integrator.du isa StaticArray)
```

Comment:
will u and `du` ever be different types?

Reply:
Unsure. I copied the format from https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/src/initialize_dae.jl#L432